src/kdc/kdc_authdata.c

```diff
@@ -373,10 +373,12 @@ fetch_kdb_authdata(krb5_context context, unsigned int flags,
         return (ret == KRB5_PLUGIN_OP_NOTSUPP) ? 0 : ret;
 
     /* Add the KDB authdata to the ticket, without copying or filtering. */
-    ret = merge_authdata(context, db_authdata,
-                         &enc_tkt_reply->authorization_data, FALSE, FALSE);
+    ret = merge_authdata(context, enc_tkt_reply->authorization_data,
```

Comment:
Changes here should probably have a comment.

Reply:
I will probably follow up this PR with a cleanup for merge_authdata().  Having allocation behavior dependent on a flag is difficult for static analysis tools (and humans) to reason about, so I will probably split the helper into a "merge without copying" variant (which nulls the list it invalidates on success, like we decided k5_add_pa_data_element() should do) and a "merge with copying" wrapper.

(Side note: there is a relatively hefty function named concat_authorization_data() which

Reply:
Right, this code can use some cleanup, I'll leave the comment to Greg.